docs/modules/usage/cloud/cloud-api.md

```diff
@@ -0,0 +1,168 @@
+  -H "Content-Type: application/json" \
+  -d '{
+    "initial_user_msg": "Check whether there is any incorrect information in the README.md file and send a PR to fix it if so.",
+    "repository": "https://github.com/yourusername/your-repo"
+  }'
+```
+</details>
+
+<details>
+<summary>Python (with requests)</summary>
+
+```python
+import requests
+
+api_key = "YOUR_API_KEY"
+url = "https://app.all-hands.dev/api/conversations"
+
+headers = {
+    "Authorization": f"Bearer {api_key}",
+    "Content-Type": "application/json"
+}
+
+data = {
+    "initial_user_msg": "Check whether there is any incorrect information in the README.md file and send a PR to fix it if so.",
+    "repository": "https://github.com/yourusername/your-repo"
+}
+
+response = requests.post(url, headers=headers, json=data)
+conversation = response.json()
+
+print(f"Conversation Link: https://app.all-hands.dev/conversations/{conversation['id']}")
+print(f"Status: {conversation['status']}")
+```
+</details>
+
+<details>
+<summary>TypeScript/JavaScript (with fetch)</summary>
+
+```typescript
+const apiKey = "YOUR_API_KEY";
+const url = "https://app.all-hands.dev/api/conversations";
+
+const headers = {
+  "Authorization": `Bearer ${apiKey}`,
+  "Content-Type": "application/json"
+};
+
+const data = {
+  initial_user_msg: "Check whether there is any incorrect information in the README.md file and send a PR to fix it if so.",
+  repository: "https://github.com/yourusername/your-repo"
+};
+
+async function startConversation() {
+  try {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: headers,
+      body: JSON.stringify(data)
+    });
+
+    const conversation = await response.json();
+
+    console.log(`Conversation Link: https://app.all-hands.dev/conversations/${conversation.id}`);
+    console.log(`Status: ${conversation.status}`);
+
+    return conversation;
+  } catch (error) {
+    console.error("Error starting conversation:", error);
+  }
+}
+
+startConversation();
+```
+
+</details>
+
+#### Response
+
+The API will return a JSON object with details about the created conversation:
+
+```json
+{
+  "status": "ok",
+  "conversation_id": "abc1234",
+}
+```
+
+### Retrieving Conversation Status
+
+You can check the status of a conversation by making a GET request to the conversation endpoint.
+
+#### Endpoint
+
+```
+GET https://app.all-hands.dev/api/conversations/{conversation_id}
+```
+
+#### Example
+
+<details>
+<summary>cURL</summary>
+
+```bash
+curl -X GET "https://app.all-hands.dev/api/conversations/{conversation_id}" \
+  -H "Authorization: Bearer YOUR_API_KEY"
+```
+</details>
+
+#### Response
+
+```json
+{
+  "conversation_id":"abc1234",
+  "title":"Update README.md",
+  "created_at":"2025-04-29T15:13:51.370706Z",
+  "last_updated_at":"2025-04-29T15:13:57.199210Z",
+  "status":"RUNNING",
+  "selected_repository":"https://github.com/yourusername/your-repo",
+  "trigger":"gui"
+}
+```
+
+## Rate Limits
```

Comment:
Similarly having a section for `Authentication error` would be helpful

1. Either you provided the wrong repo name
2. Or you don't have access to the repo